### DISCUSSION TITLE
ZFS_INITRD_ADDITIONAL_DATASETS seems to do nothing (Debian 11/Devuan 4)

### DISCUSSION THREAD

**Discussion by hyperlogos** [openzfs/zfs#14748] [Q&A] [answered]
What boot option[s] is/are ZFS_INITRD_ADDITIONAL_DATASETS meant to cause to be inserted into my command line? It seems to do nothing. I am using Devuan 4, which is based on Debian 11. https://openzfs.github.io/openzfs-docs/Getting%20Started/Debian/Debian%20Bullseye%20Root%20on%20ZFS.html suggests I put any sets needed for booting into that variable in /etc/default/zfs

**Comment by rincebrain** [comment, marked as answer]
Why do you think it causes parameters to be injected into your command line?

https://github.com/openzfs/zfs/blob/c71fe716401f6919068f84b389dcd1b7ec2b8b0e/contrib/initramfs/scripts/zfs#L977

It is intended to, unless I'm mistaken, cause them to be explicitly mounted in addition to / before pivot_root or whatever it's called nowadays.

**Comment by hyperlogos** [comment]
I hadn't the foggiest. Thanks for the link. I will explore further to see
where the breakdown is occurring.

**Comment by hyperlogos** [comment]
Welp the problem was due to the script not even being used due to a problem with a backported package. Got the normal version and all is well. Thanks for the pointer to the file, with that I was able to figure out I wasn't using it
